helpers/scaffold_json.cr

```diff
@@ -2,17 +2,44 @@ require "compiler/crystal/syntax"
 require "json"
 
 class TestCase
-  property code, name
-
-  def initialize(name : String, code : String)
+  property name : String
+  property source_code : Array(String)
+  property snippet : String
+  property start_location : Crystal::Location?
+  property end_location : Crystal::Location?
+
+  def initialize(
+    name : String,
+    source_code : Array(String),
+    snippet : String,
+    start_location : Crystal::Location?,
+    end_location : Crystal::Location?
+  )
     @name = name
-    @code = code
+    @source_code = source_code
+    @snippet = snippet
+    @start_location = start_location
+    @end_location = end_location
+  end
+
+  def test_code : String
+    start_location = self.start_location
+    end_location = self.end_location
```

Comment:
assigned to local variable so that crystal compiler can narrow the type inference for later usage